packages/synapse-react-client/src/components/Markdown/MarkdownSynapse.tsx

```diff
@@ -156,7 +156,9 @@ const MarkdownSynapse: MarkdownSynapseComponent = class MarkdownSynapse extends
         event.preventDefault()
         // handle table of contents widget
         const idOfContent = anchor.getAttribute('data-anchor')
-        const goTo = this.markupRef.current!.querySelector(`#${idOfContent}`)
+        const goTo = this.markupRef.current!.querySelector(
+          `[id="${idOfContent}"]`,
+        )
```

Comment:
In portals, there was a case where the `id` could contain a space--in those cases, `querySelector('#some id')` won't work because it ends up querying for a descendent of `#some` with tag `id`. This new query fixes that edge case.

The component that had this bug is changed in a PR coming later, but I decided to search for other usages and found this case, so I'm not sure if this code ever led to any bugs in the wild.